batchupload.php

```diff
@@ -0,0 +1,306 @@
+
+$PAGE->set_url($baseurl);
+
+if ($courseid == $SITE->id && $series) {
+    $redirecturl = new moodle_url('/blocks/opencast/overview_videos.php', ['series' => $series,
+        'ocinstanceid' => $ocinstanceid, ]);
+} else {
+    $redirecturl = new moodle_url('/blocks/opencast/index.php', ['courseid' => $courseid, 'ocinstanceid' => $ocinstanceid]);
+}
+
+require_login($courseid, false);
+
+// Check if the setting is on.
+if (empty(get_config('block_opencast', 'batchuploadenabled_' . $ocinstanceid))) {
+    throw new moodle_exception('batchupload_errornotenabled', 'block_opencast', $redirecturl);
+}
+
+// Use block context for this page to ignore course file upload limit.
+$pagecontext = upload_helper::get_opencast_upload_context($courseid);
+$PAGE->set_context($pagecontext);
+
+$PAGE->set_pagelayout('incourse');
+$PAGE->set_title(get_string('batchupload', 'block_opencast'));
+$PAGE->set_heading(get_string('pluginname', 'block_opencast'));
+$PAGE->navbar->add(get_string('pluginname', 'block_opencast'), $redirecturl);
+$PAGE->navbar->add(get_string('batchupload', 'block_opencast'), $baseurl);
+
+// Capability check.
+if ($courseid == $SITE->id) {
+    // If upload initiated from overview page, check that capability is given in specific course or ownership.
+    if (!$series) {
+        redirect(new moodle_url('/blocks/opencast/overview_videos.php', ['ocinstanceid' => $ocinstanceid, 'series' => null]),
+            get_string('addvideonotallowed', 'block_opencast'), null,
+            notification::NOTIFY_ERROR);
+    }
+
+    $records = $DB->get_records('tool_opencast_series', ['series' => $series, 'ocinstanceid' => $ocinstanceid]);
+    $haspermission = false;
+    foreach ($records as $record) {
+        $cc = context_course::instance($record->courseid, IGNORE_MISSING);
+        if ($cc && has_capability('block/opencast:addvideo', $cc)) {
+            $haspermission = true;
+            break;
+        }
+    }
+
+    if (!$haspermission) {
+        // Check if series is owned by user.
+        $apibridge = apibridge::get_instance($ocinstanceid);
+        $ocseries = $apibridge->get_series_by_identifier($series);
+        if (!$ocseries) {
+            redirect(new moodle_url('/blocks/opencast/overview_videos.php', ['ocinstanceid' => $ocinstanceid,
+                'series' => $series, ]),
+                get_string('connection_failure', 'block_opencast'), null,
+                notification::NOTIFY_ERROR);
+        }
+
+        if (!$apibridge->is_owner($ocseries->acl, $USER->id, $SITE->id)) {
+            redirect(new moodle_url('/blocks/opencast/overview_videos.php', ['ocinstanceid' => $ocinstanceid,
+                'series' => $series, ]),
+                get_string('addvideonotallowed', 'block_opencast'), null,
+                notification::NOTIFY_ERROR);
+        }
+    }
+} else {
+    $coursecontext = context_course::instance($courseid);
+    require_capability('block/opencast:addvideo', $coursecontext);
```

Comment:
In case that series has specific value is only when the form is submitted via POST, therefore you will never be able to post a form with any arbitrary series (unless you are a hacker :D)

Therefore, we leave this as it is!

Reply:
We also want to be protected against any type of unwanted behaviour, which is why it is necessary to include the check.

Reply:
Done in the last commit, please wait until all test are completed